### PR TITLE
Add OAuth2 scope support

### LINKED ISSUE #5: Oauth Access Token Scopes
### Feature Request
To add the ability to generate access tokens with Oauth scopes https://datatracker.ietf.org/doc/html/rfc6749#section-3.3

Looks like a simple enough issue, wouldn't mind opening a PR if you believe this project should support it.

### PR DESCRIPTION
**Pull request by lionello**
Fixes #5 

First time contributing and new to bun/hono, so let me know if something is out of style. Particularly not sure about usage of `test.each` for testing with and without scopes.

The OAuth2 spec is pretty thin wrt `scope`. In particular, the corner case seem to differ between implementations. For this PR, I've considered the absence of `scope` different from the empty scope, ie. `&scope=`.

Since the resource server will have to check the issued scopes, I've added `scopes` to the JWT, iff `scope` was specified in the auth request.

### CODE REVIEW

File: packages/openauth/src/client.ts [hunk 1]
```diff
@@ -38,22 +38,22 @@
  *
  * @packageDocumentation
  */
+import type { v1 } from "@standard-schema/spec"
```
Comment (lionello) [inline packages/openauth/src/client.ts:41]: These got reordered because of `Organize Imports`. Let me know if you want me to revert this part.

File: .changeset/new-donuts-tap.md
Comment (lionello) [inline .changeset/new-donuts-tap.md:1]: Not sure if this is right. Haven't used this before either.
